src/containers/ConversionForm/index.tsx

```diff
@@ -61,15 +64,35 @@ const ConversionForm = ({
     conversionProcessingData,
     setError,
     receiveFileToConvert,
+    conversionStatus,
 }: ConversionProps): JSX.Element => {
     const [fileToConvert, setFileToConvert] = useState<UploadFile>();
     const [engineSelected, setEngineSelected] = useState<boolean>(false);
+    const [serverDownModalOpen, setServerIsDownModalOpen] =
+        useState<boolean>(false);
     const [isProcessing, setIsProcessing] = useState<boolean>(false);
     const [fileTypeErrorModalOpen, setFileTypeErrorModalOpen] = useState(false);
 
-    const toggleModal = () => {
+    // On load, configure controller and check if server is healthy
+    useEffect(() => {
+        // calling this with null payload configures the controller and checks server health
+        receiveFileToConvert(null);
```

Comment:
I kind of like the idea of having a separate `configureController` prop that gets passed in? It could still be the same base method getting passed in (`configureController={() => receiveFileToConvert(null)}` in the parent) but I feel like this behavior should either be explicitly documented in the arguments or it should just be a separate prop.

Reply:
This isn't a good way to use logics/actions. If you find yourself doing this, you should either change the action that trigger the logic, or write a new action just to trigger the logic. This is breaking the point of having all these very specifically named actions that just do one thing.

Reply:
so after talking with Megan we're actually going to pull this back out into a separate action `CREATE_FILE` is going to become `INITIALIZE_FILE_CONVERSION` and we're going to do it there.